src/geo/transform.js

```diff
@@ -305,6 +317,40 @@ class Transform {
         this._cameraZoom = this._zoomFromMercatorZ(terrainElevation + height);
     }
 
+    sampleAverageElevation(): number {
+        if (!this._elevation) return 0;
+
+        const elevationSamplePoints = [
+            [0.5, 0.2],
+            [0.3, 0.5],
+            [0.5, 0.5],
+            [0.7, 0.5],
+            [0.5, 0.8]
+        ];
+
+        const horizon = this.horizonLineFromTop();
+
+        let elevationSum = 0.0;
+        let weightSum = 0.0;
+        if (this._elevation) {
+            for (let i = 0; i < elevationSamplePoints.length; i++) {
+                const pt = new Point(
+                    elevationSamplePoints[i][0] * this.width,
+                    horizon + elevationSamplePoints[i][1] * (this.height - horizon)
+                );
+                const hit = this._elevation.pointCoordinate(pt);
+                if (!hit) continue;
+
+                const weight = 1 / Math.hypot(hit[0] - this._camera.position[0], hit[1] - this._camera.position[1]);
```

Comment:
nit: Maybe make a named variable `distanceToHit` for `Math.hypot(hit[0] - this._camera.position[0], hit[1] - this._camera.position[1])`  for clariy, it's not immediately clear what this represents when you first glance at it.